### DISCUSSION TITLE
Add CORS_HOSTS option to Docker environment variaables

### DISCUSSION THREAD

**Discussion by lgaudreau** [mealie-recipes/mealie#746] [Feature Request]
If this is possible with openresty or caddy, could a CORS_HOSTS environment variable be added to allow hosts for API calls from a specific domain?

I've set up Mealie in Docker, used Pihole to create a DNS hostname for it then a forwarder in NGINX Proxy Manager which receives the hostname requests and forwards them to the docker container and port.  This is all part of getting it working with the Homer container / page.

With this set up, the Homer page can't make API calls to Mealie, with the error 'CORS header “Access-Control-Allow-Origin” missing'.

I've had no luck with adding a custom header in NGINX Proxy Manager to allow this, with very weird results - there's no good example of the correct headers to add, the ones I've found that work the best remove the CORS errors but add http 404 errors (no idea why those are happening after adding in custom headers).

**Comment by hay-kot** [comment]
Please use the [feature request template](https://github.com/hay-kot/mealie/issues/317) when submitting a request

Although, I don't believe FastAPI uses CORS by default and I have not explicitly enabled it so I'm not sure this has to do with Mealie.
